WordPress/Classes/Networking/MediaRequestAuthenticator.swift

```diff
@@ -0,0 +1,247 @@
+import Foundation
+
+fileprivate let photonHost = "i0.wp.com"
+
+extension URL {
+    /// Whether the URL is hosted at WPCom.
+    ///
+    func isHostedAtWPCom() -> Bool {
+        return host?.hasSuffix(".wordpress.com") ?? false
+    }
```

Comment:
Duplicated API (see `URL.hasWordPressDotComHostname`)

Reply:
Yeah, this one is intentional since I didn't want  to add noise to this PR.  This change is coming up in a follow-up PR (prolly tomorrow).

Reply:
See: https://github.com/wordpress-mobile/WordPress-iOS/pull/13770/files#diff-3a39681aafac124c93f4b4d4c13a3e5bL25

Reply:
Actually, you're right... the method I was talking about is a third one doing the same thing. :)

I'll unify these two in this PR and unify the third method in my upcoming PRs.

Reply:
Done!